core/Azure.Mcp.Core/src/Commands/CommandFactory.cs

```diff
@@ -115,7 +113,7 @@ private void RegisterCommandGroup()
                 throw error;
             }
 
-            if (!existingGroups.Add(area.Name))
+            if (!existingAreaNames.Add(area.Name))
             {
                 var matchingAreaTypes = _serviceAreas
                     .Where(x => x.Name == area.Name)
```

Comment:
Just a nit here, `existingAreaNames` is case-insensitive on finding duplicate groups but this `Where` LINQ filter isn't. This might lead to a hard to debug invalid case at some point.